src/main/java/io/strimzi/kafka/access/KafkaAccessReconciler.java

```diff
@@ -4,35 +4,108 @@
         this.kubernetesClient = kubernetesClient;
+        final Base64.Encoder encoder = Base64.getEncoder();
+        commonSecretData.put(TYPE_SECRET_KEY, encoder.encodeToString(TYPE_SECRET_VALUE.getBytes(StandardCharsets.UTF_8)));
+        commonSecretData.put(PROVIDER_SECRET_KEY, encoder.encodeToString(PROVIDER_SECRET_VALUE.getBytes(StandardCharsets.UTF_8)));
+        commonSecretLabels.put(Utils.MANAGED_BY_LABEL_KEY, Utils.KAFKA_ACCESS_LABEL_VALUE);
     }
 
     @Override
-    public UpdateControl<KafkaAccess> reconcile(KafkaAccess kafkaAccess, Context context) {
-        return UpdateControl.noUpdate();
+    public UpdateControl<KafkaAccess> reconcile(final KafkaAccess kafkaAccess, final Context context) {
+        final String kafkaAccessName = kafkaAccess.getMetadata().getName();
+        LOGGER.info("Reconciling KafkaAccess " + kafkaAccessName);
+        final Map<String, String> data  = new HashMap<>(commonSecretData);
+        kubernetesClient
+            .secrets()
+            .inNamespace(kafkaAccess.getMetadata().getNamespace())
+            .createOrReplace(
+                    new SecretBuilder()
+                        .withType(SECRET_TYPE)
+                        .withNewMetadata()
+                        .withName(kafkaAccessName)
+                        .withLabels(commonSecretLabels)
+                        .withOwnerReferences(
+                                new OwnerReferenceBuilder()
+                                    .withApiVersion(kafkaAccess.getApiVersion())
+                                    .withKind(kafkaAccess.getKind())
+                                    .withName(kafkaAccessName)
+                                    .withUid(kafkaAccess.getMetadata().getUid())
+                                    .withBlockOwnerDeletion(false)
+                                    .withController(false)
+                                    .build()
+                        )
+                        .endMetadata()
+                        .withData(data)
+                        .build()
+            );
+        final KafkaAccessStatus status = Optional.ofNullable(kafkaAccess.getStatus())
+                .orElse(new KafkaAccessStatus());
+        status.setBinding(new BindingStatus(kafkaAccessName));
+        kafkaAccess.setStatus(status);
+        return UpdateControl.updateStatus(kafkaAccess);
     }
 
     @Override
-    public List<EventSource> prepareEventSources(EventSourceContext<KafkaAccess> context) {
-        return Collections.emptyList();
+    public List<EventSource> prepareEventSources(final EventSourceContext<KafkaAccess> context) {
+        LOGGER.info("Preparing event sources");
+        final List<EventSource> eventSources = new ArrayList<>();
+        eventSources.add(new InformerEventSource<>(kubernetesClient,
+                Kafka.class,
+                kafka -> Utils.getKafkaAccessResourceIDsForKafkaInstance(context.getPrimaryCache().list(), kafka)
```

Comment:
Not aware the logic, and avoid list if possible, just generating the ResourceID from labels/specs/status/ownerReference, but if not possible it's fine.